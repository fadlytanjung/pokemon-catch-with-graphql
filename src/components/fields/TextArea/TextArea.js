import React from 'react';
import PropTypes from 'prop-types';
import styles from './styles.scoped.css';

export default function TextArea(props) {
  const { className, input, inputProps, meta } = props;
  const { dirty, error, touched } = meta;
  const classes = [
    styles.root,
    error && (dirty || touched) && styles.error,
    className,
  ].filter(Boolean).join(' ');

  return (
    <>
      <label className={styles.label}>{inputProps.label}</label>
      <textarea className={classes} {...input} {...inputProps} />
      {error && (dirty || touched) && <small className={styles.error}>{error}</small>}
    </>
  );
}

TextArea.defaultProps = {
  className: '',
  input: {},
  inputProps: {},
  meta: {},
};

TextArea.propTypes = {
  className: PropTypes.string,
  input: PropTypes.object,
  inputProps: PropTypes.object,
  meta: PropTypes.object,
};
