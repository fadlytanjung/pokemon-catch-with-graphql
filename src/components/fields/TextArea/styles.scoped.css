.root {
  background-color: #FFFFFF;
  border: 0.0625rem solid #cccccc;
  border-radius: 0.5rem;
  box-shadow: none;
  color: var(--light-text-color);
  display: block;
  font-size: 0.9rem;
  font-weight: 400;
  margin-top: 0;
  margin-bottom: 0;
  min-height: 6rem;
  outline: none;
  padding: 0.75rem;
  resize: vertical;
  transition: border-color 0.15s ease-in-out, box-shadow 0.15s ease-in-out;
  width: 100%;
}

.root:not(.error):focus {
  border-color: #808080;
  box-shadow: 0 0 0 0.0625rem #808080;
  color: #333435;
}

.root:disabled {
  background-color: #e5e5e5 !important;
  color: #999999 !important;
}

.root::placeholder {
  color: #b2b4b5;
}

.root.error {
  border-color: #ee4c24;
}

.root + small.error {
  color: #ee4c24;
  display: block;
  font-size: 0.75rem;
  margin-top: 0.25rem;
}

.label {
  display: inline-block;
  margin-bottom: 0.5rem;
  text-transform: uppercase;
  letter-spacing: 0.0625rem;
  color: #999999;
  font-weight: 400;
  font-size: 0.75rem;
}
