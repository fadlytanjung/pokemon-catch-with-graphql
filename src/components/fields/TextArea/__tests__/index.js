import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import TextArea from '../TextArea';

describe('src/components/fields/TextArea', () => {
  test('render', () => {
    const shallow = new ShallowRenderer();
    const tree = shallow.render(<TextArea />);
    expect(tree).toMatchSnapshot();

    const result = TextArea({
      ...TextArea.defaultProps,
      input: { value: 'test' },
      inputProps: { label: 'abc' },
    });
    const label = result.props.children[0].props.children;
    expect(label).toBe('abc');
  });

  test('render error', () => {
    const meta = { error: 'error', touched: true };
    const shallow = new ShallowRenderer();
    const tree = shallow.render(<TextArea meta={meta} />);
    expect(tree).toMatchSnapshot();
  });
});
