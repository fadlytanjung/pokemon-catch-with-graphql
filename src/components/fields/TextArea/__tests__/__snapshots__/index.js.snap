// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/fields/TextArea render 1`] = `
<React.Fragment>
  <label />
  <textarea
    className=""
  />
</React.Fragment>
`;

exports[`src/components/fields/TextArea render error 1`] = `
<React.Fragment>
  <label />
  <textarea
    className=""
  />
  <small>
    error
  </small>
</React.Fragment>
`;
